src/runtime/dict.cpp

```diff
@@ -1119,12 +1141,14 @@ void setupDict() {
 
     dict_cls->tp_hash = PyObject_HashNotImplemented;
     dict_cls->tp_compare = (cmpfunc)dict_compare;
+    dict_cls->tp_richcompare = dict_richcompare;
 
     dict_cls->giveAttr("__len__", new BoxedFunction(FunctionMetadata::create((void*)dictLen, BOXED_INT, 1)));
     dict_cls->giveAttr("__new__", new BoxedFunction(FunctionMetadata::create((void*)dictNew, UNKNOWN, 1, true, true)));
     dict_cls->giveAttr("__init__", new BoxedFunction(FunctionMetadata::create((void*)dictInit, NONE, 1, true, true)));
     dict_cls->giveAttr("__repr__", new BoxedFunction(FunctionMetadata::create((void*)dictRepr, STR, 1)));
 
+    dict_cls->giveAttr("__cmp__", new BoxedFunction(FunctionMetadata::create((void*)dict_richcompare, UNKNOWN, 2)));
```

Comment:
I think this line should be removed -- `__cmp__`  is supposed to mirror tp_compare not tp_richcompare.  Also we don't need to add these attributes manually; calling `dict_cls->freeze()` later will add them automatically for us.